examples/frost.c

```diff
@@ -0,0 +1,330 @@
+#define THRESHOLD 3
+
+struct signer_secrets {
+    secp256k1_keypair keypair;
+    secp256k1_frost_share agg_share;
+    secp256k1_frost_secnonce secnonce;
+    unsigned char seed[32];
+};
+
+struct signer {
+    secp256k1_xonly_pubkey pubkey;
+    secp256k1_pubkey pubshare;
+    secp256k1_frost_pubnonce pubnonce;
+    secp256k1_frost_session session;
+    secp256k1_frost_partial_sig partial_sig;
+    secp256k1_pubkey vss_commitment[THRESHOLD];
+    unsigned char vss_hash[32];
+    unsigned char pok[64];
+};
+
+ /* Create a key pair and store it in seckey and pubkey */
+int create_keypair(const secp256k1_context* ctx, struct signer_secrets *signer_secrets, struct signer *signer) {
+    unsigned char seckey[32];
+    while (1) {
+        if (!fill_random(seckey, sizeof(seckey))) {
+            printf("Failed to generate randomness\n");
+            return 1;
+        }
+        if (secp256k1_keypair_create(ctx, &signer_secrets->keypair, seckey)) {
+            break;
+        }
+    }
+    if (!secp256k1_keypair_xonly_pub(ctx, &signer->pubkey, NULL, &signer_secrets->keypair)) {
+        return 0;
+    }
+    return 1;
+}
+
+int create_vss(const secp256k1_context* ctx, struct signer_secrets *signer_secrets, struct signer *signer) {
+    /* Create random seed. It is absolutely necessary that the seed be unique
+     * for every distributed key generation session. */
+    if (!fill_random(signer_secrets->seed, sizeof(signer_secrets->seed))) {
+        return 0;
+    }
+    /* Create VSS commitment */
+    if (!secp256k1_frost_vss_gen(ctx, signer->vss_commitment, signer->pok, signer_secrets->seed, THRESHOLD)) {
+        return 0;
+    }
+    return 1;
+}
+
+/* Create shares and coefficient commitments */
+int create_shares(const secp256k1_context* ctx, struct signer_secrets *signer_secrets, struct signer *signer, secp256k1_xonly_pubkey *agg_pk) {
+    int i, j;
+    secp256k1_frost_share shares[N_SIGNERS][N_SIGNERS];
+    const secp256k1_pubkey *vss_commitments[N_SIGNERS];
+
+    for (i = 0; i < N_SIGNERS; i++) {
+        vss_commitments[i] = signer[i].vss_commitment;
+        for (j = 0; j < N_SIGNERS; j++) {
+            /* Generate a polynomial share for the participants */
+            if (!secp256k1_frost_share_gen(ctx, &shares[i][j], signer[j].vss_commitment, signer[j].pok, signer_secrets[i].seed, &signer[j].pubkey, THRESHOLD)) {
+                return 0;
+            }
+        }
+    }
+
+    /* KeyGen communication round 1: exchange shares and coefficient
+     * commitments */
+    for (i = 0; i < N_SIGNERS; i++) {
+        const secp256k1_frost_share *assigned_shares[N_SIGNERS];
+
+        /* Each participant receives a share from each participant (including
+         * themselves) corresponding to their index. */
+        for (j = 0; j < N_SIGNERS; j++) {
+            assigned_shares[j] = &shares[j][i];
+        }
+        /* Each participant aggregates the shares they received. */
```

Comment:
We might want to perform the verification of the VSS hashes of all participants inside of `share_agg` to enforce this verification requirement as a prerequisite to share aggregation.